modules/appeals_api/app/services/appeals_api/evidence_submission_request_validator.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  class EvidenceSubmissionRequestValidator
+    def initialize(nod_id, request_ssn)
+      @nod_id = nod_id
+      @request_ssn = request_ssn
+    end
+
+    def call
+      return [:error, record_not_found_error] unless notice_of_disagreement_exists?
+      return [:error, invalid_review_option_error] unless @notice_of_disagreement.accepts_evidence?
+      return [:error, invalid_veteran_id_error] unless ssn_match?
+
+      [:ok, {}]
```

Comment:
I think these guard clauses make it clear what is going on here! 👍   You could encapsulate the `@notice_of_disagreement.accepts_evidence?` into it's own method and then all the error states would look the same...

I have a personal preference for `if` over `unless`, but don't think you have to change it unless the rest of the team has similar feelings.

Reply:
I think unless is a clear, straight-to-the-point solution here. Please let me know if you have suggestions for refactoring :)